mdd/python/translation/openconfig_xe/xe_qos.py

```diff
@@ -0,0 +1,281 @@
+                36:'af42', 38:'af43', 40:'cs5', 46:'ef', 48:'cs6', 56:'cs7', 0:'default'}
+
+def xe_qos_program_service(self, nso_props) -> None:
+    """
+    Program service
+    """
+
+    device_cdb = nso_props.root.devices.device[nso_props.device_name].config
+
+    # Forwarding-groups
+    """
+    Configure forwarding-groups
+    """
+    for fg in nso_props.service.oc_qos__qos.forwarding_groups.forwarding_group:
+        device_cdb.ios__policy_map.create(fg.name)
+    if len(nso_props.service.oc_qos__qos.forwarding_groups.forwarding_group) == 0:
+        if device_cdb.ios__policy_map:
+            device_cdb.ios__policy_map.delete()
+
+    # Class-map
+    """
+    Configure classifiers
+    """
+    list_cmap = []
+    for c_map in nso_props.service.oc_qos__qos.classifiers.classifier:
+        pmap_cmap = {}
+        # Configure class-map
+        device_cdb.ios__class_map.create(c_map.name)
+        for t_map in c_map.terms.term:
+            pmap_cmap[t_map.actions.config.target_group] = c_map.name
+            # Configure "match ip dscp"
+            if t_map.conditions.ipv4.config.protocol == 4:
+                # Configure multiple dscp statements
+                if t_map.conditions.ipv4.config.dscp_set:
+                    for new_ip_dscp in t_map.conditions.ipv4.config.dscp_set:
+                        new_ip_dscp = modify_dscp(new_ip_dscp)
+                        device_cdb.ios__class_map[c_map.name].prematch = 'match-all'
+                        device_cdb.ios__class_map[c_map.name].match.ip.dscp.create(new_ip_dscp)
+                # Configure single dscp statements
+                elif t_map.conditions.ipv4.config.dscp:
+                    new_ip_dscp = modify_dscp(t_map.conditions.ipv4.config.dscp)
+                    device_cdb.ios__class_map[c_map.name].prematch = 'match-all'
+                    device_cdb.ios__class_map[c_map.name].match.ip.dscp.create(
+                        new_ip_dscp)
+            # Configure "match dscp"
+            else:
+                # Configure multiple dscp statements
+                if t_map.conditions.ipv4.config.dscp_set:
+                    for new_dscp in t_map.conditions.ipv4.config.dscp_set:
+                        new_dscp = modify_dscp(new_dscp)
+                        device_cdb.ios__class_map[c_map.name].prematch = 'match-all'
+                        device_cdb.ios__class_map[c_map.name].match.dscp.create(new_dscp)
+                # Configure single dscp statements
+                elif t_map.conditions.ipv4.config.dscp:
+                    new_dscp = modify_dscp(t_map.conditions.ipv4.config.dscp)
+                    device_cdb.ios__class_map[c_map.name].prematch = 'match-all'
+                    device_cdb.ios__class_map[c_map.name].match.dscp.create(
+                        new_dscp)
+        list_cmap.append(pmap_cmap)
+    if len(nso_props.service.oc_qos__qos.classifiers.classifier) == 0:
+        if device_cdb.ios__class_map:
```

Comment:
Minor, but you can just combine these two if statements since you're checking both for truthy.
```suggestion
       if (len(nso_props.service.oc_qos__qos.classifiers.classifier) == 0
           and device_cdb.ios__class_map):
```

Reply:
Updated.